src/thor/astar_bss.cc

```diff
@@ -271,27 +270,19 @@ AStarBSSAlgorithm::GetBestPath(valhalla::Location& origin,
 
   // Initialize the origin and destination locations. Initialize the
   // destination first in case the origin edge includes a destination edge.
-  uint32_t density = SetDestination(graphreader, destination);
+  SetDestination(graphreader, destination);
   SetOrigin(graphreader, origin, destination);
 
   // Find shortest path
   uint32_t nc = 0; // Count of iterations with no convergence
                    // towards destination
   std::pair<int32_t, float> best_path = std::make_pair(-1, 0.0f);
-  size_t total_labels = 0;
+  size_t n = 0;
   while (true) {
     // Allow this process to be aborted
-    size_t current_labels = edgelabels_.size();
-    if (interrupt &&
-        total_labels / kInterruptIterationsInterval < current_labels / kInterruptIterationsInterval) {
+    if (interrupt && (++n % kInterruptIterationsInterval) == 0) {
       (*interrupt)();
     }
-    total_labels = current_labels;
-
-    // Abort if max label count is exceeded
-    if (total_labels > max_label_count_) {
-      return {};
-    }
 
```

Comment:
but dont we still want this? as i was saying there are two types of aborting, using the interrupt which is telling us the client said hey i dont care about this request anymore, and then the other kind where we say ok we searched hard enough that we now give up

Reply:
I mentioned it somewhere: it’s not used at all. We still do what you’re saying, but it’s done with this variable. It’s this “nc” a bit further up.